tests/integration/replication.tcl

```diff
@@ -752,3 +752,60 @@ test {replicaof right after disconnection} {
         }
     }
 }
+
+test {Kill rdb child process if its dumping RDB is not useful} {
+    start_server {tags {"repl"}} {
+        set slave1 [srv 0 client]
+        start_server {} {
+            set slave2 [srv 0 client]
+            start_server {} {
+                set master [srv 0 client]
+                set master_host [srv 0 host]
+                set master_port [srv 0 port]
+                for {set i 0} {$i < 10} {incr i} {
+                    $master set $i $i
+                }
+                # Generating RDB will cost 1s(10 * 100ms)
+                $master config set rdb-key-save-delay 100000
+                $master config set repl-diskless-sync no
+                $master config set save ""
+
+                $slave1 slaveof $master_host $master_port
+                $slave2 slaveof $master_host $master_port
+
+                # Wait for starting child
+                wait_for_condition 50 100 {
+                    [s 0 rdb_bgsave_in_progress] == 1
+                } else {
+                    fail "rdb child didn't start"
+                }
+
+                # Slave1 disconnect with master
+                $slave1 slaveof no one
+                # Shouldn't kill child since another slave wait for rdb
+                after 100
+                assert {[s 0 rdb_bgsave_in_progress] == 1}
+
+                # Slave2 disconnect with master
+                $slave2 slaveof no one
+                # Should kill child
+                after 100
+                assert {[s 0 rdb_bgsave_in_progress] == 0}
```

Comment:
this wait and sleep can be replaced with a wait_for_condition loop. we can set it to a relatively short timeout (100 or 200ms), but in most cases it'll exit a lot faster.
setting it for a short timeout reassures us that the `after 100` above, and the `after 200` below are usually long enough.

Reply:
@oranagra  if  we set a short timeout(100 or 200ms), should we worry about time sensitivity for valgrind.
how about 100 max retries  and 10ms delay between retries for wait_for_condition loop if we want it faster

Reply:
@ShooterIT i've seen both times were the tcl code freezes (on a busy machine) while redis is getting more advance, and the opposite too.
so this 100ms sleep can in theory sleep for over a second, and wake up only after the fork already exited. that's why i prefer to stretch time of the save delay (adding more keys or increasing the sleep of each one).

for this specific comment, we expect the fork to be killed immediately (usually after 10ms), so i rather not wait the whole 100ms. and at the same time, change the max timeout to some 200 or 500 so that the test is less time sensitive in case the reaping of the child PID (checkChildrenDone) is delayed for some reason.
so i guess something like `wait_for_condition(20, 10)` would be better than `after 100`.
i generally like to avoid constant sleeps as much as i can.

Reply:
Got it, thank you very much. I started learning TCL this year, just because I also want to add test cases for my PR.